test/clojure-mode-indentation-test.el

```diff
@@ -705,6 +705,11 @@ x
     "#?@(:clj  [2]
     :cljs [2])")
 
+  (when-aligning-it "should work correctly when margin comments appear after multi-line, non-terminal sexps"
+    "(let [:x (+ 2 3
```

Comment:
This seems an odd example, I really don't recall ever seeing an aligned function invocation.

Hashmaps, `case`s, etc are more frequent targets.

Also, this piece of code would look exactly the same for the aligned and not aligned cases, so one can't really tell if you're affecting alignment at all with the proposed changes.

I'd suggest choosing a variety of examples, and also, add ` ; comment`s at different positions, to get really good coverage.

Reply:
I think alignment of `let` bindings is common. I made the strange decision to use keywords for the left-hand side of the bindings, which I will definitely update.

I will update to add examples for `hashmap` and `case`. I plan to still use a single test with a single form, but I will add all three cases within the form. I think that's what you have in mind.

I will also update so that the alignment isn't trivial; I can see how that might be confusing, but I also what to make clear that it is besides the point. The point is that when you align code that is already aligned correctly, it is getting mangled ([see the attached issue](https://github.com/clojure-emacs/clojure-mode/issues/608)). This test does fail if you undo the regexp change I made.

Reply:
Thanks! Appreciated.

Simply for record, I still don't see a particular alignment in the example, neither for the `let` itself or the function invocations:

![image](https://user-images.githubusercontent.com/1162994/152226144-2f078f02-6ef4-4e33-9ad1-ae00f3721676.png)

I'm aware that `let` alignment is a thing but it's better shown e.g. here:

![image](https://user-images.githubusercontent.com/1162994/152226493-1bb6b29c-840d-4ec2-a175-34c633034e03.png)

...Just in case, indentation != alignment.

btw, this initial newline I added in both of my examples would make the tests more readable. I think it would be welcome if you can update all examples to include it.

(I haven't tried running the tests like that, might require removing the initial newline programatically if present)